_plugins/filters/meta_util.rb

```diff
@@ -0,0 +1,86 @@
+# traverses the page and site objects to find the most appropriate
+# meta data tags based on a predetermined priority
+
+module Jekyll
+  module CRDS
+    class MetaUtil
+      # Priority:
+      # 1) meta image
+      # 2) page image
+      # 3) page background image
+      # 4) first image found in page content
+      # 5) site variable's image
+      # Don't love it -- some pretty brittle checks based on naming conventions established in media
+      # method for getting the url for the <meta name="image" content="{{ meta_image }}"> tag in the head of a SSG html page
+      def self.get_meta_image(page_meta_image, page_image, page_bg_image, page_content, site_image)
+
+        # returns the first image url in the array that has a value
+        image_url = [
+          page_meta_image,
+          page_image,
+          page_bg_image,
+          search_for_first_image_url_in_page_content(page_content), #invoking the regex this way returns the FIRST match
+          site_image
+        ].find{|image_under_review| has_value?(image_under_review)}.to_s.strip
```

Comment:
Checkout  [Array.compact](https://ruby-doc.org/core-2.2.0/Array.html#method-i-compact)

```ruby
image_url = args.compact.shift.to_s.strip
```

Reply:
@tcmacdonald Couldn't apply this one directly without giving up some functionality, but nevertheless, new tool in my Ruby belt.